provisioning/transport/amqp/src/amqp.ts

```diff
@@ -125,11 +126,23 @@ export class Amqp extends EventEmitter implements X509ProvisioningTransport, Tpm
           _onEnter: (request, callback) => {
             /*Codes_SRS_NODE_PROVISIONING_AMQP_16_002: [The `registrationRequest` method shall connect the AMQP client with the certificate and key given in the `auth` parameter of the previously called `setAuthentication` method.]*/
             /*Codes_SRS_NODE_PROVISIONING_AMQP_16_012: [The `queryOperationStatus` method shall connect the AMQP client with the certificate and key given in the `auth` parameter of the previously called `setAuthentication` method. **]**]*/
-            const config: AmqpBaseTransportConfig = {
+            let config: AmqpBaseTransportConfig = {
               uri: this._getConnectionUri(request),
               sslOptions: this._x509Auth,
               userAgentString: ProvisioningDeviceConstants.userAgent
             };
+            if (this._sasToken) {
+              /*Codes_SRS_NODE_PROVISIONING_AMQP_06_002: [** The `registrationRequest` method shall connect the amqp client, if utilizing the passed in sas token from setSasToken, shall in the connect options set the username to:
+                ```
+                <scopeId>/registrations/<registrationId>
+                ```
+                and shall set the password to the passed in sas token.
+                ] */
+              config.policyOverride = {
+                username: request.idScope + '/registrations/' + request.registrationId,
+                password: this._sasToken
+              };
```

Comment:
huh that policyOverride thing is a leak of the rhea library in higher layers. shouldn't happen.

Reply:
I'm not seeing your point here.  the policy override mechanism was part of the base transport config forever.  This is passing down a username and password to the base client.  That is absolutely what would be needed to perform a sasl plain.  This doesn't particularly call out any rhea detail.

---
In reply to: [231364570](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231364570) [](ancestors = 231364570)